### DISCUSSION TITLE
PSA: SFTPGO_COMMON__PROXY_PROTOCOL=1 can cause slow downs if not disabled

### DISCUSSION THREAD

**Discussion by matthewlenz** [drakkan/sftpgo#1066] [General]
I am mainly using SFTPgo through a proxy solution for the SSH and HTTP based communication.  I had a need to allow FTP which the user connects directly to the server (rather than using the proxy).  I discovered that SFTPGO_COMMON__PROXY_PROTOCOL=1 was causing a large delay (4-5 seconds for a response) for any connections/operations over FTP.    I don't see the same latency for non-FTP communication directly to the server (non-proxied).
 SFTPGO_FTPD__BINDINGS__0__APPLY_PROXY_CONFIG=false remedies this issue.  I don't really know that it's a bug as it does make sense that not having the proxy header could add some latency (I just wasn't expecting that much).  Hopefully this helps someone.

**Comment by drakkan** [comment]
After waiting for the delay, do you see the proxy IP as the client IP in the logs or the real IP of the connecting client?

**Reply by matthewlenz** [reply]
I do get the correct IP because I'm not connecting through the proxy for FTP.  For non-FTP proxy or direct connectivity are all working perfectly and showing the correct IP (client/real) in the logs/connections.  Everything works.  It's just that FTP is slow when connecting directly.  The slowness is fixed with SFTPGO_FTPD__BINDINGS__0__APPLY_PROXY_CONFIG=false.  That's completely fine for my situation though because I have no intention of proxying FTP connections.

**Reply by drakkan** [reply]
> I do get the correct IP because I'm not connecting through the proxy for FTP. For non-FTP proxy or direct connectivity are all working perfectly and showing the correct IP (client/real) in the logs/connections. Everything works. It's just that FTP is slow when connecting directly. The slowness is fixed with SFTPGO_FTPD__BINDINGS__0__APPLY_PROXY_CONFIG=false. That's completely fine for my situation though because I have no intention of proxying FTP connections.

Slowness is generally a timeout waiting for the proxy header that SFTPGo expects based on its configuration. This does not happen for SFTP because the initial data exchange exceeds the maximum number of bytes that are expected to read the proxy header
